src/comparator.ml

```diff
@@ -0,0 +1,8 @@
+
+type 'a t = 'a -> 'a -> TotalOrder.t
+
+let make (compare : 'a -> 'a -> TotalOrder.t) (this : 'a) (that : 'a): TotalOrder.t =
```

Comment:
The intent of that 'make' function is to take a traditional ocaml compare function which returns an int and convert it into an Immutable.re compatible Comparator function which returns an Ordering. 
```
type t 'a = 'a => 'a => Ordering.t;

let make (compare: 'a => 'a => int) (this: 'a) (that: 'a): Ordering.t => {
  let cmp = compare this that;

  if (cmp > 0) Ordering.greaterThan
  else if (cmp < 0) Ordering.lessThan
  else Ordering.equal;
};
```